src/transport/SecureSession.h

```diff
@@ -158,6 +158,8 @@ class SecureSession : public Session, public ReferenceCounted<SecureSession, Sec
 
     bool RequireMRP() const override { return GetPeerAddress().GetTransportType() == Transport::Type::kUdp; }
```

Comment:
Pre-existing, but this makes no sense.  UDP _allows_ MRP, but does not require it.  Follow-up issue for fixing that?

But fundamentally, I don't think we need a `RequireTCP` function.  We should have `AllowsMRP` (false for group sessions, true for unicast UDP sessions, false for unicast BTP/TCP sessions; it's the current RequireMRP, but named clearly), and just test that as needed.

Reply:
Yes, I was also questioning the naming and just went with the existing model. I will change it to AllowsMRP() in this PR which makes more sense. I want to have a RequiresTCP() as I was planning to use it for other purposes on the session, as an easy way to determine what approach to adopt on the session handling side.

Reply:
That's fine, but none of the code being changed for this PR should need to use RequiresTCP.

And maybe RequiresTCP should be called TransportIsTCP() or something?

Reply:
Made it IsTransportTCP()